src/test/java/StringAddCalculatorTest.java

```diff
@@ -0,0 +1,73 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import util.Tokenizer;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class StringAddCalculatorTest {
+    StringAddCalculator stringAddCalculator;
+
+    @Test
+    public void splitAndSum_null_또는_빈문자() {
+        int result = stringAddCalculator.splitAndSum(null);
+        assertThat(result).isEqualTo(0);
+
+        result = stringAddCalculator.splitAndSum("");
+        assertThat(result).isEqualTo(0);
+    }
+
+    @Test
+    public void splitAndSum_숫자하나() throws Exception {
+        int result = stringAddCalculator.splitAndSum("1");
+        assertThat(result).isEqualTo(1);
+    }
+
+    @Test
+    public void splitAndSum_쉼표구분자() throws Exception {
+        int result = stringAddCalculator.splitAndSum("1,2");
+        assertThat(result).isEqualTo(3);
+    }
+
+    @Test
+    public void splitAndSum_쉼표_또는_콜론_구분자() throws Exception {
+        int result = stringAddCalculator.splitAndSum("1,2:3");
+        assertThat(result).isEqualTo(6);
+    }
+
+    @Test
+    public void splitAndSum_custom_구분자() throws Exception {
+        int result = stringAddCalculator.splitAndSum("//;\n1;2;3");
+        assertThat(result).isEqualTo(6);
+    }
+
+    @Test
+    public void splitAndSum_negative() throws Exception {
+        assertThatThrownBy(() -> stringAddCalculator.splitAndSum("-1,2,3"))
+                .isInstanceOf(RuntimeException.class);
+    }
+
+    @Nested
+    class StringManipulatiton {
+        @Test
+        void customTokenizer() {
+            String target  = "//;\\n1;2;3";
+            assertThat(target.substring(target.indexOf("//")+2, target.indexOf("\\"))).isEqualTo(";");
+            assertThat(target.substring(target.indexOf("\\")+2)).isEqualTo("1;2;3");
```

Comment:
```suggestion
            assertThat(target.substring(target.indexOf("//")+2, target.indexOf("\\")))
                        .isEqualTo(";");
            assertThat(target.substring(target.indexOf("\\")+2))
                        .isEqualTo("1;2;3");
```

사소해보이지만, 가독성을 위해 한 줄에 점이 한개 찍힐 수 있도록, 엔터를 아끼지 말아주세요